assembly-syntax/src/sema/context.rs

```diff
@@ -171,3 +180,18 @@ impl AnalysisContext {
     #[cfg(not(feature = "std"))]
     fn emit_warnings(self) {}
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Converts the digest obtained from the [`Blake3_256::hash`] procedure into the [`Word`].
+fn blake3_digest_to_word(blake3_digest: Blake3Digest<32>) -> Word {
```

Comment:
I would probably make this function more generic. Maybe something like:
```rust
fn hash_string_to_word(value: impl Into<&str>) -> Word {
    ...
}
```
Also, should probably move this into the `utils` module of `miden-core` and export it from there.

Lastly, can we use this function for hashing error messages too? (I think we use the same methodology there). We don't need to change how error messages work, but somewhere, to get the error code, we hash the error string using BLAKE3, I believe.